js-jasmine/src/agedbrie.js

```diff
@@ -0,0 +1,19 @@
+
+class AgedBrie {
+  constructor (items = []) {
+    this.items = items
+  }
+
+  update () {
+    for (let i = 0; i < this.items.length; i++) {
+      if (this.items[i].name === 'Aged Brie') {
+        this.items[i].sellIn -= 1
+        if (this.items[i].quality < 50) {
```

Comment:
I like how throughout the classes you can see that you have really tried to refactor and haven't just rewritten a lot of the code!